src/modules/linked_data/wordlift_linked_data.php

```diff
@@ -247,33 +213,37 @@ function wl_save_entities( $entities, $related_post_id = null ) {
  * Save the specified data as an entity in WordPress. This method only create new entities. When an existing entity is
  * found (by its URI), then the original post is returned.
  *
- * @param array $entity_properties , associative array containing:
- * string 'uri' The entity URI.
- * string 'label' The entity label.
- * string 'main_type_uri' The entity type URI.
- * string 'description' The entity description.
- * array 'type_uris' An array of entity type URIs.
- * array 'images' An array of image URLs.
- * int 'related_post_id' A related post ID.
- * array 'same_as' An array of sameAs URLs.
+ * @param array $entity_data, associative array containing: 
+ * string 'uri'             The entity URI.
+ * string 'label'           The entity label.
+ * string 'main_type'       The entity type URI.
+ * array  'type'            An array of entity type URIs.
+ * string 'description'     The entity description.
+ * array  'images'          An array of image URLs.
+ * int    'related_post_id' A related post ID.
+ * array  'same_as'         An array of sameAs URLs.
  *
  * @return null|WP_Post A post instance or null in case of failure.
  */
-function wl_save_entity( $entity_properties ) {
-
-	$uri             = $entity_properties['uri'];
-	$label           = $entity_properties['label'];
-	$type_uri        = $entity_properties['main_type_uri'];
-	$description     = $entity_properties['description'];
-	$entity_types    = $entity_properties['type_uris'];
-	$images          = $entity_properties['images'];
-	$related_post_id = $entity_properties['related_post_id'];
-	$same_as         = $entity_properties['same_as'];
-
-	// Avoid errors due to null.
-	if ( is_null( $entity_types ) ) {
-		$entity_types = array();
-	}
+function wl_save_entity( $entity_data ) {
+	
+	$uri              = $entity_data['uri'];
+	$label            = $entity_data['label'];
+	$type_uri         = $entity_data['main_type'];
+	$entity_types     = isset( $entity_data['type'] ) ? $entity_data['type'] : array();
+	$description      = $entity_data['description'];
+	$images           = isset( $entity_data['image'] ) ?
```

Comment:
this is a bit hard to read:
    isset( $entity_data['image'] ) ? ( is_array( $entity_data['image'] ) ? $entity_data['image'] : array( $entity_data['image'] ) ) : array();

can we make more readable?